### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by Sajjon**
Fix some typos

### CODE REVIEW

**Comment by amartini51** [comment]
@swift-ci Please test.

**Comment by Sajjon** [comment]
@amartini51 That was quick! :D Yes sorry, I found some more typos and wanted to push those.... Maybe force pushing is frowned upon? But I wanted to avoid many commits with same message "fix typo".

I will mark it as ready when I'm done!

**Comment by Sajjon** [comment]
@amartini51 ready! I think I found all typos - or rather Xcode `Check Spelling Now` did, but one has to go through each file individually - an improvement opportunity for Xcode! :)

**Comment by Sajjon** [comment]
@swift-ci Please test.

File: Sources/TSPL/TSPL.docc/LanguageGuide/Concurrency.md [hunk 1]
```diff
@@ -822,7 +822,7 @@ actor TemperatureLogger {
 
 
 @Comment {
-  - test: `actors, actors-implicity-sendable`
+  - test: `actors, actors-implicitly-sendable`
```
Comment (amartini51) [inline Sources/TSPL/TSPL.docc/LanguageGuide/Concurrency.md:825]: Confirmed that there isn't another test with this name, either spelled correctly or misspelled.

(If there were, we'd need to make sure joining the code listings was correct and that the tests still passed.)

File: Sources/TSPL/TSPL.docc/ReferenceManual/Declarations.md [hunk 2]
```diff
@@ -693,7 +693,7 @@ func sum<T: Sequence>(_ sequence: T) -> Int where T.Element == Int {
 
 
 @Comment {
-  - test: `typealias-in-prototol`
+  - test: `typealias-in-protocol`
```
Comment (amartini51) [inline Sources/TSPL/TSPL.docc/ReferenceManual/Declarations.md:696]: Confirmed that there isn't another test with this name, either spelled correctly or misspelled.

(If there were, we'd need to make sure joining the code listings was correct and that the tests still passed.)

File: Sources/TSPL/TSPL.docc/ReferenceManual/Declarations.md [hunk 3]
```diff
@@ -3266,7 +3266,7 @@ extension Array: Loggable where Element: MarkedLoggable { }
 }
 
 @Comment {
-  - test: `types-cant-have-multiple-implict-conformances`
+  - test: `types-cant-have-multiple-implicit-conformances`
```
Comment (amartini51) [inline Sources/TSPL/TSPL.docc/ReferenceManual/Declarations.md:3269]: Confirmed that there isn't another test with this name, either spelled correctly or misspelled.

(If there were, we'd need to make sure joining the code listings was correct and that the tests still passed.)

File: Sources/TSPL/TSPL.docc/ReferenceManual/Attributes.md [hunk 4]
```diff
@@ -1418,7 +1418,7 @@ that varies between the conditional branches.
 
 For example, the code below defines a simple result builder
 that builds an array of integers.
-This code defines `Compontent` and `Expression` as type aliases,
+This code defines `Component` and `Expression` as type aliases,
```
Comment (amartini51) [inline Sources/TSPL/TSPL.docc/ReferenceManual/Attributes.md:1421]: This is the one typo fix that will change the published content — the others are in `@Comment` blocks.

**Review by amartini51 — APPROVED** [review]
These typo fixes look good to me.  Left a couple notes of the checking I did when reading through the diff.

**Pull request merged**